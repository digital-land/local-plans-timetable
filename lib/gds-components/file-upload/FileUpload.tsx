/**
 * GDS component: https://design-system.service.gov.uk/components/file-upload/
 */

interface FileUploadProps {
  label: string;
  onChange: (value: File) => void;
  accept?: string;
}

export const FileUpload = ({
  label,
  onChange,
  accept = ".csv",
}: FileUploadProps) => {
  const handleChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
    if (e.target.files) {
      onChange(e.target.files[0]);
    }
  };

  return (
    <div className="govuk-form-group">
      <label className="govuk-label">{label}</label>
      <input
        className="govuk-file-upload"
        type="file"
        accept={accept}
        onChange={handleChange}
      />
    </div>
  );
};
