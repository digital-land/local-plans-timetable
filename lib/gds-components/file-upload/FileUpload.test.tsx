import "@testing-library/jest-dom";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";

import { FileUpload } from "./FileUpload";

const labelText = "Upload a file";
const onChange = jest.fn();

describe("FileUpload", () => {
  it("calls onChange with the right value when input changes", async () => {
    const { asFragment } = render(
      <FileUpload label={labelText} onChange={onChange} />
    );

    const file = new File([], "test.csv", {
      type: ".svg",
    });

    const input = screen.getByLabelText(labelText);

    await waitFor(() =>
      fireEvent.change(input, {
        target: { files: [file] },
      })
    );

    expect(asFragment()).toMatchSnapshot();
    expect(onChange).toHaveBeenCalledWith(file);
  });
});
