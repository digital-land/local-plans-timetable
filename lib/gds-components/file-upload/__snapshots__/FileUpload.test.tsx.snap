// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload calls onChange with the right value when input changes 1`] = `
<DocumentFragment>
  <div
    class="govuk-form-group"
  >
    <label
      class="govuk-label govuk-label--m"
      for="file-upload"
    >
      Upload a file
    </label>
    <input
      accept=".csv"
      class="govuk-file-upload"
      id="file-upload"
      type="file"
    />
  </div>
</DocumentFragment>
`;
